dumux-precice/couplingadapter.cc

```diff
@@ -310,111 +179,63 @@ bool CouplingAdapter::isCoupledEntity(const int faceID) const
     return indexMapper_.isDumuxIdMapped(faceID);
 }
 
-size_t CouplingAdapter::getIdFromName(const std::string &dataName) const
+std::string CouplingAdapter::createKeyFromName(
+    const precice::string_view &meshName,
+    const precice::string_view &dataName) const
 {
     assert(wasCreated_);
-    const auto it = std::find(dataNames_.begin(), dataNames_.end(), dataName);
-    if (it == dataNames_.end()) {
-        throw(std::runtime_error(" Error! Name of data not found! "));
+    std::string combinedKey;
+    int length = meshName.size() + 1 + dataName.size();
+    for (int i = 0; i < length; i++) {
+        if (i < meshName.size())
+            combinedKey += meshName[i];
+        else if (i == meshName.size())
+            combinedKey += ":";
+        else
+            combinedKey += dataName[i - meshName.size() - 1];
```

Comment:
How are we using this logic?

Reply:
> How are we using this logic?
I try to combine the `meshName` and `dataName` to a `string` key for the data map with the formation: `meshName:DataName`.

Reply:
Okay. Lets name `combinedKey` appropriately then, perhaps `meshAndDataKey` or `dataAndMeshKey`, whichever you think is more appropriate.